src/rsa/padding.rs

```diff
@@ -159,3 +160,179 @@ pkcs1_digestinfo_prefix!(
 pkcs1_digestinfo_prefix!(
     SHA512_PKCS1_DIGESTINFO_PREFIX, 64, 9,
     [ 0x60, 0x86, 0x48, 0x01, 0x65, 0x03, 0x04, 0x02, 0x03 ]);
+
+
+/// PSS Padding as described in https://tools.ietf.org/html/rfc3447#section-9.1.
+/// It generates a random salt equal in length to the output of the specified
+/// digest algorithm and uses MGF1 with that digest algorihtm as the mask
+/// generating function.
+pub struct PSS {
+    digest_alg: &'static digest::Algorithm,
+}
+
+// Fixed prefix used in the computation of PSS encoding and verification.
+const PSS_PREFIX_ZEROS: [u8; 8] = [0u8; 8];
+
+impl Verification for PSS {
+    // PSS verification as specified in
+    // https://tools.ietf.org/html/rfc3447#section-9.1.2
+    fn verify(&self, msg: untrusted::Input, encoded: untrusted::Input,
+              public_modulus_len_in_bits: usize)
+              -> Result<(), error::Unspecified> {
+        // Number of bytes required to encode message. The maximum length is
+        // given by the length of the public modulus in bits minus 1.
+        let em_len = 1 + (public_modulus_len_in_bits - 2) / 8;
+        assert_eq!(encoded.len(), em_len);
```

Comment:
I don't think this is correct. Let's say we have a 2049 bit modulus. Then `encoded.len() == (2049 + 7) / 8`, which is 257. However, `emBits == modLen - 1` == 2048, and `emLen == ceil(emBits / 8)`, which is 256. Thus we have `encoded.len() != em_len` here.

I have a patch that fixes this.